### ISSUE TITLE
Preserve logs from kubernetes_manifests_deployer_job

### ISSUE THREAD

**Issue by NimJay** [GoogleCloudPlatform/terraform-ecommerce-microservices-on-gke#29]
* Currently, the logs of the `kubernetes_deployer_job` last as long as the Pod (of the Job) lasts.
* That is, as soon as the Job completes [`deploy_k8s_resources.sh`](https://github.com/GoogleCloudPlatform/terraform-ecommerce-microservices-on-gke/blob/3cc833e5d0c90e88be7b3e000b2a66cc8a6bb7fd/kubernetes_manifests_deployer/deploy_k8s_resources.sh), the Pod used for the Job disappears and the logs from that Pod disappear.
* This makes debugging errors in the `kubernetes_deployer_job` extremely difficult.
* We need to figure out a way to preserve the logs.

### Current Workaround

* The current workaround for obtaining the logs is to run something like the following toward the end of `terraform apply` or _just_ after `terraform apply` completes (i.e., while the Job's Pod is still alive):
```sh
# Connect to the config cluster.
gcloud container clusters get-credentials my-cluster-config-1 \
    --region us-west1 \
    --project MY_PROJECT_ID

# Get the name of the Pod that ran kubernetes_deployer_job.
kubectl get pods

# Get the logs.
kubectl get logs NAME_OF_POD_FROM_ABOVE_COMMAND >> logs_from_kubernetes_deployer_job
```

**Comment by bourgeoisor** [comment]
Double check if `kubectl logs job/<job_name>` gives you want you want

**Comment by NimJay** [comment]
Hey @bourgeoisor, excellent suggestion!

I tried `kubectl logs job/kubernetes-manifests-deployer-job` on a _successful_ deployment, and I got the full logs.
The Pod was also still listed when I ran `kubectl get pods`:
```
NAME                                      READY   STATUS      RESTARTS   AGE
kubernetes-manifests-deployer-job-hw4fl   0/1     Completed   0          19h
```

I have *not* tried `kubectl logs job/kubernetes-manifests-deployer-job` on a *failing* deployment though (but will likely find out in [b/279866344](http://b/279866344) where we have a *failing* deployment.
Nonetheless, I will close this issue for now.
